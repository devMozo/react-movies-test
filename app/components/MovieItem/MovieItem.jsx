import React from 'react';
import { Link } from 'react-router-dom';
import './MovieItem.scss';
import { MOVIES_IMG_PATH } from '../../constants/config';
import PropTypes from 'prop-types';

const MovieItem = ({ id, title, posterPath, overview }) => {
    return (
        <li className="MovieItem" data-cy="MovieItem">
            <header className="MovieItem__header">
                <picture className="MovieItem__pic">
                    <img src={MOVIES_IMG_PATH + posterPath} alt={title} />
                </picture>
                <h2> {title} </h2>
            </header>
            <p> {overview} </p>
            <Link to={`/movie/${id}`} data-cy="MovieItem-link">
                {' '}
                Go to the movie!{' '}
            </Link>
        </li>
    );
};

MovieItem.propTypes = {
    id: PropTypes.number,
    title: PropTypes.string,
    posterPath: PropTypes.string,
    overview: PropTypes.string,
};

export default React.memo(MovieItem);
