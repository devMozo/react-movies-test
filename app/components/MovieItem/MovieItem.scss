.MovieItem {
    text-align: left;
    background-color: white;
    box-shadow: 0 0 10px #ccc;
    margin: 40px 0;
    padding: 40px;
    box-sizing: border-box;
    list-style: none;

    .MovieItem__header {
        & > * {
            display: inline-block;
            vertical-align: middle;
        }

        .MovieItem__pic img {
            width: 100px;
            height: 100px;
            object-fit: cover;
            margin-right: 20px;
        }

        h4 {
            color: #333;
        }
    }

    p {
        color: #666;
    }

    a {
        display: inline-block;
        padding: 20px 50px;
        margin-top: 10px;
        border: 1px solid #aaa;
        color: #333;
        transition: 0.3s;
        text-decoration: none;

        &:hover {
            border: 1px solid #333;
            background-color: #333;
            color: white;
        }
    }
}
