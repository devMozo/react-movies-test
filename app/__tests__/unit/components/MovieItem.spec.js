import React from 'react';
import { shallow } from 'enzyme';
import MovieItem from '../../../components/MovieItem/MovieItem';

describe('<MovieItem />', () => {
    const PROPS = { id: 1234, title: 'mocked_title', posterPath: 'poster.jpg', overview: 'mocked_overview' };
    let component;

    beforeEach(() => {
        component = shallow(<MovieItem {...PROPS} />);
    });

    it('should render correctly in "debug" mode', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render correctly each field', () => {
        const image = component.find('.MovieItem__pic img');
        const title = component.find('.MovieItem__header h2');
        const overview = component.find('.MovieItem > p');

        expect(image.prop('src')).toContain(PROPS.posterPath);
        expect(title.contains(PROPS.title)).toBeTruthy();
        expect(overview.contains(PROPS.overview)).toBeTruthy();
    });
});
