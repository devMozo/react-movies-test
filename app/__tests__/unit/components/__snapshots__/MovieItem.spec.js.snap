// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItem /> should render correctly in "debug" mode 1`] = `
<li
  className="MovieItem"
>
  <header
    className="MovieItem__header"
  >
    <picture
      className="MovieItem__pic"
    >
      <img
        alt="mocked_title"
        src="https://image.tmdb.org/t/p/w500/poster.jpg"
      />
    </picture>
    <h2>
       
      mocked_title
       
    </h2>
  </header>
  <p>
     
    mocked_overview
     
  </p>
  <Link
    to="/movie/1234"
  >
     Go to the movie! 
  </Link>
</li>
`;
